### PR TITLE
Add tax report comparison filters

### LINKED ISSUE #924: Analytics: Tax Reports Filters
**Design Post**: p6riRB-3dT-p2
**Mock API**: [https://href.li/?https://app.swaggerhub.com/apis/peterfabian/wc-v3-api/1.0.0#/default/get_reports_taxes_stats](get_reports_taxes_stats)
**REST API Post**: p7bje6-15p-p2

The tax report needs to support the ability to filter by Tax Codes. Akin to being able to filter the orders report by one or many products - a similar advanced filter needs to be created and implemented on the Tax Report.

### PR DESCRIPTION
**Pull request by joshuatf**
Fixes #924 

Adds in the comparison filter for taxes per the designs p6riRB-3dT-p2

### Screenshots
screen shot 2018-12-05 at 2 37 05 pm

### Detailed test instructions:

1.  Make sure you've added at least 2 tax rates under `wp-admin/admin.php?page=wc-settings&tab=tax&section=standard`
2.  Open `wp-admin/admin.php?page=wc-admin#/analytics/taxes`
3.  Change the filter method to "Comparison" under "Show:"
4.  Select a couple tax rates based on the the tax code (follows the WC format of {country}-{state}-{name}).
5.  Click "Compare."
6.  Note the change in filter params in the URL.

The query is passed to the report table, but will not affect data until #844 is implemented.

### CODE REVIEW

File: client/analytics/report/taxes/utils.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+/** @format */
+/**
+ * External dependencies
+ */
+
+export function getTaxCode( tax ) {
+	return [ tax.country, tax.state, tax.name ].join( '-' );
```
Comment (Aljullu) [inline client/analytics/report/taxes/utils.js]: When `tax.country` or `tax.state` are blank this shows several dashes in a row:

image

How does it sound filtering the strings before joining them?
```suggestion
	return [ tax.country, tax.state, tax.name ].filter( Boolean ).join( '-' );
```

**Review by Aljullu — COMMENTED** [review]
This is testing well and code looks good. Just a couple of comments.

I'm not sure if it's related to this PR or we should open a separate issue, but I noticed apostrophes appear with a backslash before them:
image

File: client/analytics/report/taxes/utils.js (hunk 1 above)
Comment (joshuatf) [reply client/analytics/report/taxes/utils.js]: Thanks!  Seems like a smart move.  Also trimmed here and modified to match the tax codes in WC, adding 'TAX' as the name if none is defined and appending the priority to the end of the tax code.

**Comment by joshuatf** [comment]
Thanks for reviewing @Aljullu !  Made the changes to the tax code mentioned.

As for the escaping issue, opened an issue in WC as this appears to be an existing issue on current tax reports as well.  https://github.com/woocommerce/woocommerce/issues/22149

**Review by Aljullu — APPROVED** [review]
It works as expected and code looks good to me! :rocket:

**Pull request merged**